### DISCUSSION TITLE
MissingArgument  error

### DISCUSSION THREAD

**Discussion by gherlt** [ImageMagick/ImageMagick#1984] [General] [answered]
Hi all,
I am a newby, I installed it on my Win10 64 PC.
For a start, I am trying to batch-change the canvas size (bigger for all) with image centered.
I re-read the comments on the options three times, but it is not clear to me what is wrong with this command line:
magick *.jpg -extent 3600x3200 -gravity center
the following error appears
magick: MissingArgument `-gravity' at CLI arg 4 @ fatal/magick-cli.c/ProcessCommandOptions/447.

if I change th eorder
magick *.jpg -gravity center -extent 3600x3200
it is
magick: MissingArgument `-extent' at CLI arg 4 @ fatal/magick-cli.c/ProcessCommandOptions/447.

What I am doing wrong ?
I do see one strange thing in this error:  `-gravity'
it has a different delimiter on the start to the one on the end.
Might be that ?
Thank you

**Comment by dlemstra** [comment, marked as answer]
It looks like you want to perform a `mogrify` operation and the ordor of your command is not correct. You should change your command to this:

```
magick mogrify -gravity center -extent 3600x3200 *.jpg
```

**Reply by gherlt** [reply]
Thank you ! That works.
I made a search on imagemagick and found the details.
After reading that, just one comment then, for the content of the website.
The description of "command-line" section does not mention the use of the commands "mogrify" nor "convert".
So, when I read the command-line section, I still would not see what I have been doing wrong.
May be a small addiiton would be helpful for others to follow.

Nevertheless, thank you very much !

**Reply by gherlt** [reply]
sorry, one detail. no error messages now, but the -gravity center did not work correctly.
I changed the option order to 
-extent 3600x3200 -gravity center
and now all is as I need it.
